### ISSUE TITLE
should improve how to configure rsync__exclude

### ISSUE THREAD

**Issue by hoatle** [teracyhq/dev#209]
on:

```
      {
        "type":"rsync",
        "host":"./workspace",
        "guest":"/home/vagrant/workspace",
        "rsync__exclude": [".git", ".idea/", "node_modules/", "bower_components/", ".npm/"]
      },
```

To override rsync__exclude, currently, we must copy paste a lot into vagrant_config_override.json:

```
{
  "vm": {
    "synced_folders":[ //see: http://docs.vagrantup.com/v2/synced-folders/index.html
      // {
      //   "type":"virtual_box", // virtual_box, "nfs", "rsync" or "smb", incase smb mount failed,
      //                         // ssh to vm and install cifs-utils allow vm mount smb directory
      //                         // mount.cifs needs to be used with dir_mode file_mode instead of dmode fmode
      //                         // Windows 7 needs powershell 3.0 to create smb share on windows
      //                         // http://www.microsoft.com/en-us/download/details.aspx?id=34595
      //   "host":"./docker",
      //   "guest":"/var/lib/docker/",
      //   "mount_options":[
      //     "dmode=775",
      //     "fmode=755"
      //   ]
      // },
      //TODO: how to keep and reused existing Docker images after destroying the VM?
      // or even attaching the existing data into it, so that developers should share the Docker images
      // remove /vagrant default file sync on the VM guest machine #184
      {
        "type": "virtual_box",
        "host": ".",
        "guest": "/vagrant",
        "disabled": true
      },
      {
        "type":"rsync",
        "host":"./workspace",
        "guest":"/home/vagrant/workspace",
        "rsync__exclude": [".git", ".idea/", "node_modules/", "bower_components/", ".npm/"] // change it here
      },
      // should persist docker auth token after login success #170
      {
        "type":"virtual_box",
        "host":"./home/.docker",
        "guest":"/home/vagrant/.docker",
        "mount_options":[
          "dmode=775",
          "fmode=755"
        ]
      }
    ]
  }
}
```

This is not good.

related: #203

**Comment by hoatle** [comment]
maybe support environment variables, such as: https://www.packer.io/docs/templates/configuration-templates.html

**Comment by hoatle** [comment]
should be fixed by #239
